pkgs/racket-doc/scribblings/reference/equality.scrbl

```diff
@@ -634,6 +655,27 @@ and only access mutable data when the mode is true:
 ]
 ]
 
+Certain type-specific equality functions on mutable types,
+such as @racket[string=?], should also be avoided unless the
+struct is mutable or the mode is true.
+Type-specific equality functions on immutable types, such as
+@racket[symbol=?], are fine.
```

Comment:
Why is it fine for immutable types? The advice needs more explanation about how we're trying to ensure `chaperone-of?` doesn't break.

Reply:
A previous paragraph in the documentation attempts to explain it like this:

> Mutable structs will only use the custom equality for `equal?` and `impersonator-of?`, so that `equal-always?` and `chaperone-of?` don't change on mutation. Structs that represent mutable data should either be declared mutable, or use `equal-mode-proc` from `gen:equal-mode+hash` instead of `equal-proc` from `gen:equal+hash`, and only access mutable data when the mode is true

But that explanation isn't good enough to make it obvious that it applies to `string=?` being wrong here, like 3 codeblock-comparison examples down.

The first-pass guideline could be something like "don't access potentially-mutable data", and then exceptions like, "it's okay in `equal-proc` if the struct is declared mutable or okay in `equal-mode-proc` if the mode is true", can be separated later.

If I reorder them so the `string=?` comes first with the "don't access potentially-mutable data" guideline above it, that could be clearer.